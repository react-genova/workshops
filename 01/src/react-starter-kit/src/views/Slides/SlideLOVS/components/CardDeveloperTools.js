//@flow
import React from 'react';
import { Card, CardElement } from '../../../Card';

type Props = {
    onClick:(url:string)=>void
};

const CardDeveloperTools = ({onClick}:Props) => (
    <Card title="Developer tools" description="setting you up, boring part">
        <CardElement title="Node.js" description="the engine behind the curtain" url="https://nodejs.org/it/" onClick={onClick} />
        <CardElement title="Visual Studio Code" description="your IDE and your best friend" url="https://code.visualstudio.com/" onClick={onClick}>
            <CardElement description="Babel ES6/ES7" title="" url="" onClick={onClick}/>
            <CardElement description="ESLint" title="" url="" onClick={onClick}/>
            <CardElement description="Flow Language Support" title="" url="" onClick={onClick}/>
            <CardElement description="Path Intellisense" title="" url="" onClick={onClick}/>
            <CardElement description="Relative Path" title="" url="" onClick={onClick}/>
        </CardElement>
    </Card>
);

export default CardDeveloperTools;