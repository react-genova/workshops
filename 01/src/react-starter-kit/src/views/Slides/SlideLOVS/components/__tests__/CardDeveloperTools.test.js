//@flow
import React from 'react';
import { shallow } from 'enzyme';
import CardDeveloperTools from '../CardDeveloperTools';

describe('Card Developer Tools', () => {

    it('snapshot check', () => {
        const onClick = (url:string)=>{url.toLocaleLowerCase();};
        const component = shallow(<CardDeveloperTools onClick={onClick} />);
        expect(component).toMatchSnapshot();
    });

});