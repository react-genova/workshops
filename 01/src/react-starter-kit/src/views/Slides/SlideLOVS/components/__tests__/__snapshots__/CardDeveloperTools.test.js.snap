// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Developer Tools snapshot check 1`] = `
<Card
  description="setting you up, boring part"
  title="Developer tools"
>
  <CardElement
    description="the engine behind the curtain"
    onClick={[Function]}
    title="Node.js"
    url="https://nodejs.org/it/"
  />
  <CardElement
    description="your IDE and your best friend"
    onClick={[Function]}
    title="Visual Studio Code"
    url="https://code.visualstudio.com/"
  >
    <CardElement
      description="Babel ES6/ES7"
      onClick={[Function]}
      title=""
      url=""
    />
    <CardElement
      description="ESLint"
      onClick={[Function]}
      title=""
      url=""
    />
    <CardElement
      description="Flow Language Support"
      onClick={[Function]}
      title=""
      url=""
    />
    <CardElement
      description="Path Intellisense"
      onClick={[Function]}
      title=""
      url=""
    />
    <CardElement
      description="Relative Path"
      onClick={[Function]}
      title=""
      url=""
    />
  </CardElement>
</Card>
`;
